### ISSUE TITLE
Add mechanism for chaining formatters

### ISSUE THREAD

**Issue by dbarnett** [google/vim-codefmt#44]
For some types of files you might want to run more than one formatter, for instance sort C++ includes (which clang-format doesn't handle) and then format with clang-format. Could codefmt support configuring a list of formatters instead of a single formatter? E.g.
```vim
AutoFormatBuffer sortimports,clang-format
let b:codefmt_formatter = ['sortimports', 'clang-format']
```

**Comment by dbarnett** [comment]
Our hacky solution when this came up (for imports+formatting in java) was to define a little one-off formatter with its own formatter name that delegated and called formatter A then formatter B.

As noted in #178, you can't just invoke `:AutoFormatBuffer` twice and its docs don't explicitly explain it'll replace any previous autoformatting each time you invoke it.

**Comment by TamaMcGlinn** [comment]
I think my hacky solution is a bit easier than what you describe, since I only needed to add a function to do the delegation, instead of a whole formatter.

**Comment by dbarnett** [comment]
Yes, that's probably close to what I would've done as a one-off workaround. For our purposes on the java thing, we were solving a common issue and couldn't plop arbitrary autocmds into everyone's vimrc.

**Comment by TamaMcGlinn** [comment]
Correct, thank you! I had this because I had a different formatter which wasn't able to accept stdin, so it works on a post-write hook and then needs to write the file again:

```
autocmd BufWritePost *.ad[sb] Autoformat | noautocmd write
```
